taskchampion/taskchampion/src/server/cloud/gcp.rs

```diff
@@ -25,9 +26,17 @@ fn is_http_error<T>(query: u16, res: &std::result::Result<T, http::Error>) -> bo
 }
 
 impl GcpService {
-    pub(in crate::server) fn new(bucket: String) -> Result<Self> {
+    pub(in crate::server) fn new(bucket: String, credentialpath:String) -> Result<Self> {
```

Comment:
Since this is a pure Rust function, it should take the common Rust type, `Option<String>`, instead of special-casing the empty string. The conversion from the special-cased-empty-string to None is done in `tc_server_new_gcp`, which is a wrapper around `GcpService::new`.

Reply:
So I need to change the datatype for both `bucket` and `credentialpath`?

Reply:
No, Bucket can still be a String because it must be present.

Reply:
okay 👍🏻